File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict


def solution(clothes):
    clothes_dict = defaultdict(lambda: 0)
    for name, clothes_type in clothes:
        clothes_dict[clothes_type] += 1
    
    result = 1
    for value in clothes_dict.values():
        result *= (value + 1)
    
    return result - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const hashmap = clothes.reduce((acc, [name, type]) => {
        acc.set(type, (acc.get(type) || 0) + 1)
        return acc
    }, new Map())

    let result = 1
    hashmap.forEach((value) => {
        result *= value + 1
    })
    
    return result - 1;
}